presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -476,7 +476,15 @@ private void invokeNoMoreSplitsIfNecessary()
             // Create a registry of delete_delta directories for the partition
             for (AcidUtils.ParsedDelta delta : directory.getCurrentDirectories()) {
                 if (delta.isDeleteDelta()) {
-                    acidInfoBuilder.addDeleteDelta(delta.getPath(), delta.getMinWriteId(), delta.getMaxWriteId(), delta.getStatementId());
+                    OptionalInt statementId = OptionalInt.empty();
+
+                    // Handling different formats of Delete delta file:
+                    // 1. delete_delta_minWriteId_maxWriteID_statementId: default case
+                    // 2. delete_delta_minWriteId_maxWriteID: after minor compaction
+                    if (delta.getPath().getName().split("_").length == 5) {
```

Comment:
I guess this cannot be as simple as
```
if (delta.getStatementId() != 0) {
```

because  statement id 0 seems  valid. Is it?

If we cannot get the information out of `delta` object normally,
i would rather use reflection than duplicate the logic from
https://github.com/apache/hive/blob/468907eab36f78df3e14a24005153c9a23d62555/ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java#L937-L959

Reply:
I've also opened https://issues.apache.org/jira/browse/HIVE-24082

Reply:
Yeah, I was not able to use `getStatementId()` directly, I have changed the logic to compare the generated path from min, max, statement ID. Please let me know if it looks fine.

Thank you for filing the Hive issue.